pkg/domain/runaway.go

```diff
@@ -43,7 +45,11 @@ func (do *Domain) initResourceGroupsController(ctx context.Context, pdClient pd.
 		return nil
 	}
 
-	control, err := rmclient.NewResourceGroupController(ctx, uniqueID, pdClient, nil, rmclient.WithMaxWaitDuration(runaway.MaxWaitDuration))
+	keyspaceID := constants.NullKeyspaceID
+	if v := keyspace.GetKeyspaceIDBySettings(); v != nil {
+		keyspaceID = *v
+	}
```

Comment:
These four lines of code look quite general; can they be abstracted into a function?